src/controllers/password.resets.controller.js

```diff
@@ -0,0 +1,94 @@
+const Bcrypt = require('bcryptjs')
+const { Container } = require('typedi')
+const { validateAll } = require('indicative/validator')
+const RandomString = require('randomstring')
+const addMinutes = require('date-fns/addMinutes')
+const {
+    PASSWORD_RESETS_SERVICE,
+    PASSWORD_RESET,
+    EVENT_DISPATCHER
+} = require('../utils/constants')
+
+class PasswordResetsService {
+    constructor() {
+        this.EventEmitter = Container.get(EVENT_DISPATCHER)
+        this.PasswordResetsService = Container.get(PASSWORD_RESETS_SERVICE)
+
+        this.setNewPassword = this.setNewPassword.bind(this)
+        this.hashPassword = this.hashPassword.bind(this)
+        this.successResponse = this.successResponse.bind(this)
+        this.forgotPassword = this.forgotPassword.bind(this)
+        this.validate = this.validate.bind(this)
+        this.customMessages = this.customMessages.bind(this)
+        this.validationRules = this.validationRules.bind(this)
+        this.successResponse = this.successResponse.bind(this)
+    }
+
+    async setNewPassword(request, response) {
+        const resetToken = await this.PasswordResetsService.findToken(
+            request.params.token
+        )
+
+        if (this.PasswordResetsService.expired(resetToken.expiresAt))
+            throw new Error('Token Expired')
```

Comment:
If the token has expired, before throwing this error, be sure to delete the token. We do not want it anymore. Since we'll be deleting tokens, let's change this message to be `Invalid token`. Even when the token didn't exist, it'll say `Invalid token`. That way it just makes sense.